go/host/host.go

```diff
@@ -1030,7 +1030,11 @@ func (h *host) handleBatchRequest(encodedBatchRequest *common.EncodedBatchReques
 		return fmt.Errorf("could not decode batch request using RLP. Cause: %w", err)
 	}
 
-	batches, err := h.batchManager.GetBatches(batchRequest)
+	resolver := func(batchHash common.L2BatchHash) (*common.ExtBatch, error) {
+		return h.enclaveClient.GetBatch(batchHash)
```

Comment:
I'd just pass the `enclaveClient` into the `batchManager`.
Not sure about the use of a  lambda.